packages/perseus/src/widgets/interactive-graphs/graphs/components/angle-indicators.tsx

```diff
@@ -0,0 +1,428 @@
+ */
+const RightAngleSquare = ({
+    start: [x1, y1],
+    vertex: [x2, y2],
+    end: [x3, y3],
+}) => (
+    <MafsCssTransformWrapper>
+        <path
+            d={`M ${x1} ${y1} L ${x3} ${y3} M ${x3} ${y3} L ${x2} ${y2}`}
+            strokeWidth={0.02}
+            fill="none"
+        />
+    </MafsCssTransformWrapper>
+);
+
+const isRightPolygonAngle = (angle: number) =>
+    Math.abs(angle - Math.PI / 2) < 0.01;
+const isRightAngle = (angle: number) => Math.round(angle) === 90;
+
+/**
+ * Determines if an angle is an inside (false) or outside (true) angle.
+ * They way, we know to flip the `largeArc` and `sweepArc` flags.
+ * Uses the priciple that a ray from a point inside a polygon will intersect
+ * with an odd number of lines, while a ray from a point outside the polygon
+ * will intersect with an even number of lines.
+ * https://stackoverflow.com/questions/217578/how-can-i-determine-whether-a-2d-point-is-within-a-polygon
+ */
+export const shouldDrawArcOutside = (
+    midpoint: vec.Vector2,
+    vertex: vec.Vector2,
+    range: [Interval, Interval],
+    polygonLines: readonly CollinearTuple[],
+) => {
+    // Create a ray from the midpoint (inside angle) to the edge of the range
+    const rangeIntersectionPoint = getRangeIntersectionVertex(
+        midpoint,
+        vertex,
+        range,
+    );
+
+    let lineIntersectionCount = 0;
+
+    polygonLines.forEach(
+        (line) =>
+            linesIntersect([vertex, rangeIntersectionPoint], line) &&
+            lineIntersectionCount++,
+    );
+
+    // If the number of intersections is even, the angle is inside the polygon
+    return !isEven(lineIntersectionCount);
+};
+
+// https://stackoverflow.com/a/24392281/7347484
+// The "intersects" function in geometry doesn't seem to work for this use case
+const linesIntersect = (
+    [[a, b], [c, d]]: CollinearTuple,
+    [[p, q], [r, s]]: CollinearTuple,
+) => {
+    const determinant = (c - a) * (s - q) - (r - p) * (d - b);
+    if (determinant === 0) {
+        return false;
+    } else {
+        const lambda = ((s - q) * (r - a) + (p - r) * (s - b)) / determinant;
+        const gamma = ((b - d) * (r - a) + (c - a) * (s - b)) / determinant;
+        return 0 < lambda && lambda < 1 && 0 < gamma && gamma < 1;
+    }
+};
+
+const isEven = (n: number) => n % 2 === 0;
+
+// Exported for testing
+export const findAngle = (
+    point1: vec.Vector2,
+    point2: vec.Vector2,
+    vertex?: vec.Vector2,
+): number => {
+    if (vertex === undefined) {
+        const x = point1[0] - point2[0];
+        const y = point1[1] - point2[1];
+        if (!x && !y) {
+            return 0;
+        }
+        return (180 + (Math.atan2(-y, -x) * 180) / Math.PI + 360) % 360;
+    }
+    return findAngle(point1, vertex) - findAngle(point2, vertex);
+};
+
+function scaledPolarRad(radius: number, radians: number): vec.Vector2 {
```

Comment:
This is how the old text was being positioned, we have redone the math from scratch